src/daft-logical-plan/src/optimization/rules/reorder_joins/join_graph.rs

```diff
@@ -449,77 +442,112 @@ impl JoinGraphBuilder {
     /// Combining each of their resolved `left_on` conditions with their respective resolved `right_on` conditions produces
     /// a join edge between the relation used in the left condition and the relation used in the right condition.
     fn process_node(&mut self, plan: &LogicalPlanRef) {
-        let schema = plan.schema();
-        for (name, node, done) in &mut self.join_conds_to_resolve {
-            if !*done && schema.has_field(name) {
-                *node = plan.clone();
+        let mut cur_node = plan;
```

Comment:
I think it would be less bug prone to have

`let old_plan = plan;`
and then you can modify `plan` freely instead of referring to `cur_node`